packages/next/src/server/base-server.ts

```diff
@@ -2484,10 +2484,8 @@ export default abstract class Server<
         query: origQuery,
       })
 
-      const shouldWaitOnAllReady =
-        !supportsDynamicResponse ||
```

Comment:
we already check this in the renderer, mixing it with `shouldWaitOnAllReady` could easily fail.

e.g. when  `supportsDynamicResponse` is false, `isStaticGeneration` can be true for static generation case.
But wait all condition shouldn't be turned on, otherwise it could be unexpected dynamic rendering.